### ISSUE TITLE
Parallel building has random fails

### ISSUE THREAD

**Issue by catap** [thepowersgang/mrustc#225]
When I try to run on my system sometime build fails with error like:
```
:info:build > /opt/local/var/macports/build/_Users_catap_src_macports-ports_lang_mrustc/mrustc/work/mrustc-4bee8893e8f471d315c1d2d9535216b116cdfe23/bin/mrustc rustc-1.39.0-src/vendor/cc/src/lib.rs -o output-1.39.0/libcc-1_0_35.rlib --crate
-name cc --crate-type rlib -C emit-depfile=output-1.39.0/libcc-1_0_35.rlib.d --crate-tag 1_0_35 -g --cfg debug_assertions -O -L output-1.39.0
:info:build > /opt/local/var/macports/build/_Users_catap_src_macports-ports_lang_mrustc/mrustc/work/mrustc-4bee8893e8f471d315c1d2d9535216b116cdfe23/bin/mrustc rustc-1.39.0-src/src/build_helper/lib.rs -o output-1.39.0/libbuild_helper-0_1_0.
rlib --crate-name build_helper --crate-type rlib -C emit-depfile=output-1.39.0/libbuild_helper-0_1_0.rlib.d --crate-tag 0_1_0 -g --cfg debug_assertions -O -L output-1.39.0 --edition 2018
:info:build > /opt/local/var/macports/build/_Users_catap_src_macports-ports_lang_mrustc/mrustc/work/mrustc-4bee8893e8f471d315c1d2d9535216b116cdfe23/bin/mrustc rustc-1.39.0-src/vendor/libc/build.rs --crate-name build --crate-type bin -o out
put-1.39.0/build_libc-0_2_62_Hd_run -L output-1.39.0 -g --cfg feature=rustc-dep-of-std --cfg feature=align --cfg feature=rustc-std-workspace-core
:info:build > /opt/local/var/macports/build/_Users_catap_src_macports-ports_lang_mrustc/mrustc/work/mrustc-4bee8893e8f471d315c1d2d9535216b116cdfe23/bin/mrustc rustc-1.39.0-src/vendo
r/compiler_builtins/build.rs --crate-name build --crate-type bin -o output-1.39.0/build_compiler_builtins-0_1_18_H22_run -L output-1.39.0 -g --cfg feature=compiler-builtins --cfg feature=rustc-dep-of-std --cfg feature=core
:info:build > /opt/local/var/macports/build/_Users_catap_src_macports-ports_lang_mrustc/mrustc/work/mrustc-4bee8893e8f471d315c1d2d9535216b116cdfe23/output-1.39.0/build_libc-0_2_62_Hd_run
:info:build > /opt/local/var/macports/build/_Users_catap_src_macports-ports_lang_mrustc/mrustc/work/mrustc-4bee8893e8f471d315c1d2d9535216b116cdfe23/rustc-1.39.0-src/vendor/libc/output-1.39.0/build_compiler_builtins-0_1_18_H22_run
:info:build ESC[31mUnable to run process '/opt/local/var/macports/build/_Users_catap_src_macports-ports_lang_mrustc/mrustc/work/mrustc-4bee8893e8f471d315c1d2d9535216b116cdfe23/rustc-1.39.0-src/vendor/libc/output-1.39.0/build_compiler_builtins-0_1_18_H22_run' - File existsESC[0m
:info:build Calling /opt/local/var/macports/build/_Users_catap_src_macports-ports_lang_mrustc/mrustc/work/mrustc-4bee8893e8f471d315c1d2d9535216b116cdfe23/rustc-1.39.0-src/vendor/libc/output-1.39.0/build_compiler_builtins-0_1_18_H22_run failed (see /opt/local/var/macports/build/_Users_catap_src_macports-ports_lang_mrustc/mrustc/work/mrustc-4bee8893e8f471d315c1d2d9535216b116cdfe23/output-1.39.0/build_compiler_builtins-0_1_18_H22.txt_failed.txt for stdout)
```

Restart or removing `output-1.39.0/build_compiler_builtins-0_1_18_H22*` helps.

**Comment by thepowersgang** [comment]
Might be an issue with running with `-j` caused by bad dependency rules inside the makefile (leading to two copies of minicargo trying to build the same crates)

**Comment by thepowersgang** [comment]
Workaround is to not use `-j` with make (at least, not when doing the rustc/cargo/test builds - feel free to use it for building mrustc)

**Comment by catap** [comment]
Just for records an attempt to build local root 18909056c907d2fa07f095c613131b3b1382d0af with machine with 12 cores:
```
make -j12 all 2>/dev/null && env OPENSSL_DIR=/opt/local/libexec/openssl11 PARLEVEL=12 ./build-1.54.0.sh
```
works without an issue.

**Comment by catap** [comment]
and when I run:
```
env PARLEVEL=8 ./build-1.54.0.sh 
```
it fails as:
```
(42/114) BUILDING autocfg v1.0.1
> /Users/catap/src/mrustc/bin/mrustc rustc-1.54.0-src/vendor/autocfg/src/lib.rs -o output-1.54.0/cargo-build/libautocfg-1_0_1.rlib --crate-name autocfg --crate-type rlib -C emit-depfile=output-1.54.0/cargo-build/libautocfg-1_0_1.rlib.d --crate-tag 1_0_1 -g --cfg debug_assertions -O -L output-1.54.0 -L output-1.54.0/cargo-build
(43/114) BUILDING pkg_config from pkg-config v0.3.18
> /Users/catap/src/mrustc/bin/mrustc rustc-1.54.0-src/vendor/pkg-config/src/lib.rs -o output-1.54.0/cargo-build/libpkg_config-0_3_18.rlib --crate-name pkg_config --crate-type rlib -C emit-depfile=output-1.54.0/cargo-build/libpkg_config-0_3_18.rlib.d --crate-tag 0_3_18 -g --cfg debug_assertions -O -L output-1.54.0 -L output-1.54.0/cargo-build
> /Users/catap/src/mrustc/output-1.54.0/cargo-build/build_log-0_4_14_H20000_run
> /Users/catap/src/mrustc/output-1.54.0/cargo-build/build_crc32fast-1_2_1_H2_run
> /Users/catap/src/rust-legacy-bootstrap/vendor/log/output-1.54.0/cargo-build/build_memchr-2_4_0_H6_run
(44/114) BUILDING matches v0.1.8
> /Users/catap/src/mrustc/bin/mrustc rustc-1.54.0-src/vendor/matches/lib.rs -o output-1.54.0/cargo-build/libmatches-0_1_8.rlib --crate-name matches --crate-type rlib -C emit-depfile=output-1.54.0/cargo-build/libmatches-0_1_8.rlib.d --crate-tag 0_1_8 -g --cfg debug_assertions -O -L output-1.54.0 -L output-1.54.0/cargo-build
Unable to run process '/Users/catap/src/rust-legacy-bootstrap/vendor/log/output-1.54.0/cargo-build/build_memchr-2_4_0_H6_run' - File exists
Unable to run process '/Users/catap/src/mrustc/bin/mrustc' - No such file or directory
Calling /Users/catap/src/rust-legacy-bootstrap/vendor/log/output-1.54.0/cargo-build/build_memchr-2_4_0_H6_run failed (see /Users/catap/src/mrustc/output-1.54.0/cargo-build/build_memchr-2_4_0_H6.txt_failed.txt for stdout)
(45/114) BUILDING log v0.4.14 with features [std]
> /Users/catap/src/mrustc/bin/mrustc rustc-1.54.0-src/vendor/log/src/lib.rs -o output-1.54.0/cargo-build/liblog-0_4_14_H20000.rlib --crate-name log --crate-type rlib -C emit-depfile=output-1.54.0/cargo-build/liblog-0_4_14_H20000.rlib.d --crate-tag 0_4_14_H20000 -g --cfg debug_assertions -O -L output-1.54.0 -L output-1.54.0/cargo-build --cfg atomic_cas --cfg has_atomics --cfg feature="std" --extern cfg_if=output-1.54.0/cargo-build/libcfg_if-1_0_0.rlib
Unable to run process '/Users/catap/src/mrustc/bin/mrustc' - No such file or directory
(46/114) BUILDING crc32fast v1.2.1 with features [std]
> /Users/catap/src/mrustc/bin/mrustc rustc-1.54.0-src/vendor/crc32fast/src/lib.rs -o output-1.54.0/cargo-build/libcrc32fast-1_2_1_H2.rlib --crate-name crc32fast --crate-type rlib -C emit-depfile=output-1.54.0/cargo-build/libcrc32fast-1_2_1_H2.rlib.d --crate-tag 1_2_1_H2 -g --cfg debug_assertions -O -L output-1.54.0 -L output-1.54.0/cargo-build --cfg crc32fast_stdarchx86 --cfg feature="std" --extern cfg_if=output-1.54.0/cargo-build/libcfg_if-1_0_0.rlib
Unable to run process '/Users/catap/src/mrustc/bin/mrustc' - No such file or directory
(47/114) BUILDING typenum v1.12.0
> /Users/catap/src/mrustc/bin/mrustc rustc-1.54.0-src/vendor/typenum/src/lib.rs -o output-1.54.0/cargo-build/libtypenum-1_12_0.rlib --crate-name typenum --crate-type rlib -C emit-depfile=output-1.54.0/cargo-build/libtypenum-1_12_0.rlib.d --crate-tag 1_12_0 -g --cfg debug_assertions -O -L output-1.54.0 -L output-1.54.0/cargo-build
Unable to run process '/Users/catap/src/mrustc/bin/mrustc' - No such file or directory
BUILD FAILED
make: *** [output-1.54.0/cargo] Error 1
```
with created an empty file:
```
-rw-r--r--  1 catap  staff  0 Feb  8 19:25 output-1.54.0/cargo-build/build_memchr-2_4_0_H6.txt_failed.txt
```

**Comment by thepowersgang** [comment]
`Unable to run process '/Users/catap/src/mrustc/bin/mrustc' - No such file or directory` This is very odd. I suspect something is afoot with your install.

**Comment by catap** [comment]
@thepowersgang maybe I need to reboot machine. Good idea.

**Comment by catap** [comment]
=> I can confirm that the root cause of issue was a network mounted `~/src`

**Comment by yu-re-ka** [comment]
I am seeing the same issue as @catap, with `Unable to run process '.../mrustc/bin/mrustc' - No such file or directory` randomly appearing in the build with `-j1 PARLEVEL=10`.
Both with v0.10 and current master.
My source is not network-mounted.
Maybe re-open this?

**Comment by thepowersgang** [comment]
RAM consumption seems like a possibility - mrustc can use a LOT of ram (especially on some crates, e.g. `librustc`)

**Comment by thepowersgang** [comment]
The underlying issue here should be fixed in #316
